### ISSUE TITLE
Job run twice when timer triggers to early

### ISSUE THREAD

**Issue by timjwright** [robfig/cron#94]
I have had an instance where some jobs ran twice which looks to be cause by the timer triggering a few seconds to early.

The line (cron.go:201 on current master)
`e.Next = e.Schedule.Next(now)`
will cause a job to run again if the timer pops early.

In my case the jobs was scheduled a few days ahead - i don't have an explantation why the started a few seconds early.

I notice this was changed from:
`e.Next = e.Schedule.Next(effectiveTime)`
to prevent catch up effect in the machine sleeps.

However this new behaviour is a real problem for me.

I would suggest a check if when the timer is triggered: is now before the effective time -  if so continue and let the timer be scheduled again.  We could also mitigate by never setting timer for more than 1 hour so now is regularly reset?

However this could be an issue if it triggers before and keeps waking up to early?

Can I also ask about the v1 and v2 branches (i assume v1 is very old and should be ignored) v2 has extra features - but is missing some fixes from master....

I am going to submit a pull request

**Comment by timjwright** [comment]
I have submitted pull required - #96 which proposes a solution to this.

**Comment by ernsheong** [comment]
I'm also seeing possibility of double job behavior (emails sent twice), though I am not sure what is causing this.

**Comment by robfig** [comment]
Please try v3. I believe this was fixed in 316e2a23 by checking the activation times explicitly:

```
			case now = <-timer.C:
				now = now.In(c.location)
				// Run every entry whose next time was less than now
				for _, e := range c.entries {
					if e.Next.After(now) || e.Next.IsZero() {
						break
					}
					go c.runWithRecovery(e.Job)
					e.Prev = e.Next
					e.Next = e.Schedule.Next(now)
				}
```

**Comment by robfig** [comment]
Closing, please open a new issue if you see any problems.
